### PR TITLE
LPD-17102 Test fix for com.liferay.headless.batch.engine.[...].testPostExportTask

### PR DESCRIPTION
**Pull request by magjed4289**
https://issues.liferay.com/browse/LPD-15844

### CODE REVIEW

File: modules/apps/headless/headless-batch-engine/headless-batch-engine-test/src/testIntegration/java/com/liferay/headless/batch/engine/resource/v1_0/test/ExportTaskResourceTest.java [hunk 1]
```diff
@@ -165,6 +165,18 @@ public void testPostExportTask() throws Exception {
 		}
 	}
 
+	private String _splitClassName(String className) {
+		if (className.contains("#")) {
+			String[] classNameTaskItemDelegateName = className.split("#");
+
+			_taskItemDelegateName = classNameTaskItemDelegateName[1];
```
Comment (matijapetanjek) [inline modules/apps/headless/headless-batch-engine/headless-batch-engine-test/src/testIntegration/java/com/liferay/headless/batch/engine/resource/v1_0/test/ExportTaskResourceTest.java:172]: I wouldn't say it's the best practice that a private method __splitClassName_  also sets an instance field. 

If this was a solution to return 2 values from a single method, I would suggest to use [KeyValuePair](https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/util/KeyValuePair.java). If that is not available on the classpath simply use some Java collection like Map or even array.

**Review by matijapetanjek — CHANGES_REQUESTED** [review]

File: modules/apps/headless/headless-batch-engine/headless-batch-engine-test/src/testIntegration/java/com/liferay/headless/batch/engine/resource/v1_0/test/ExportTaskResourceTest.java (hunk 1 above)
Comment (magjed4289) [reply modules/apps/headless/headless-batch-engine/headless-batch-engine-test/src/testIntegration/java/com/liferay/headless/batch/engine/resource/v1_0/test/ExportTaskResourceTest.java:172]: thanks for the comment @matijapetanjek Do you mean to make it something like this?
	```
private Map<String, String> _splitClassName(String className) {
		Map<String, String> result = new HashMap<>();

		if (className.contains("#")) {
			String[] classNameTaskItemDelegateName = className.split("#");

			result.put("className", classNameTaskItemDelegateName[0]);
			result.put("taskItemDelegateName", classNameTaskItemDelegateName[1]);
		} else {
			result.put("className", className);
		}
```

Comment (matijapetanjek) [reply modules/apps/headless/headless-batch-engine/headless-batch-engine-test/src/testIntegration/java/com/liferay/headless/batch/engine/resource/v1_0/test/ExportTaskResourceTest.java:172]: LGTM :D

File: modules/apps/headless/headless-batch-engine/headless-batch-engine-test/src/testIntegration/java/com/liferay/headless/batch/engine/resource/v1_0/test/ExportTaskResourceTest.java [hunk 2]
```diff
@@ -187,8 +193,8 @@ private void _testPostExportTask(String className) throws Exception {
 		).build();
 
 		ExportTask exportTask = exportTaskResource.postExportTask(
-			_splitClassName(className), "jsont", null, null, null,
-			_taskItemDelegateName);
+			_splitClassName(className).get("className"), "jsont", null, null, null,
```
[the remark this answers is not in the document]
Comment (magjed4289) [reply modules/apps/headless/headless-batch-engine/headless-batch-engine-test/src/testIntegration/java/com/liferay/headless/batch/engine/resource/v1_0/test/ExportTaskResourceTest.java]: oh sorry, I just saw it, will modify now!

**Comment by magjed4289** [comment]
ci:test:relevant

**Comment by magjed4289** [comment]
ci:test:sf

**Review by matijapetanjek — APPROVED** [review]

**Comment by magjed4289** [comment]
ci:forward

**Comment by magjed4289** [comment]
ci:stop

**Comment by magjed4289** [comment]
ci:test:sf

**Comment by magjed4289** [comment]
ci:stop

**Comment by magjed4289** [comment]
ci:forward:force

**Comment by magjed4289** [comment]
ci:forward:force

**Comment by magjed4289** [comment]
resent [here](https://github.com/brianchandotcom/liferay-portal/pull/147127)

**Pull request closed without merging**